src/client/components/ProductsList/ProductsList.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import ProductsListItem from '../ProductsListItem';
+import toggle from './actions';
+
+const DISPLAY_LIMIT = 3;
+
+class ProductsList extends Component {
+  state = {
+    limit: DISPLAY_LIMIT
+  };
+
+  showMore = () => {
+    this.setState(({ limit }) => ({ limit: limit + DISPLAY_LIMIT }));
+  };
+
+  toggleItem = (id, isOrdered) => {
+    const { toggle } = this.props;
+    toggle(id, isOrdered);
+  };
+
+  render() {
+    const { isArchive, products } = this.props;
+    const { limit } = this.state;
+
+    return (
+      <Fragment>
+        {!products.length ? (
+          <div className="products__message">
+            <p>There are no items!</p>
+          </div>
+        ) : (
+          <ul className="products-list">
+            {products.slice(0, limit).map(item => (
+              <ProductsListItem
+                author={item.author}
+                id={item._id}
+                image={item.image}
+                isArchive={isArchive}
+                key={item._id}
+                name={item.name}
+                toggleItem={this.toggleItem}
+              />
+            ))}
+          </ul>
+        )}
+        {limit < products.length && (
+          <button
+            className="products__show-more"
+            onClick={this.showMore}
+            type="button"
+          />
+        )}
+      </Fragment>
+    );
+  }
+}
+
+ProductsList.propTypes = {
+  toggle: PropTypes.func,
+  isArchive: PropTypes.bool,
+  products: PropTypes.arrayOf(PropTypes.object)
```

Comment:
sort props and separate functions

Reply:
should it always be like, properties first and then down below functions?

Even if this is not in alphabetical order??